src/MongoDB.Driver.Core/Core/Clusters/CryptClientCreator.cs

```diff
@@ -65,24 +65,21 @@ private CryptClient CreateCryptClient(CryptOptions options)
 
         private CryptOptions CreateCryptOptions()
         {
-            Dictionary<KmsType, IKmsCredentials> kmsProvidersMap = null;
+            Dictionary<KmsType, KmsCredentials> kmsProvidersMap = null;
             if (_kmsProviders != null && _kmsProviders.Count > 0)
             {
-                kmsProvidersMap = new Dictionary<KmsType, IKmsCredentials>();
-                if (_kmsProviders.TryGetValue("aws", out var awsProvider))
+                kmsProvidersMap = new Dictionary<KmsType, KmsCredentials>();
+                foreach (var kmsProvider in _kmsProviders)
                 {
-                    if (awsProvider.TryGetValue("accessKeyId", out var accessKeyId) &&
-                        awsProvider.TryGetValue("secretAccessKey", out var secretAccessKey))
+                    if (!Enum.TryParse<KmsType>(kmsProvider.Key, true, out var kmsType))
```

Comment:
If we decide to defer all validation on the mongocrypt level, then this validation can be removed